### PR TITLE
TT-2687 added ability to receive cert id on remove key

### PR DESCRIPTION
**Pull request by sredxny**
## Description
On remove key, added ability to receive cert id so we can remove keys with certs.  This is a pre-requisite to run https://github.com/TykTechnologies/tyk-analytics/pull/2357 so we must merge them in tandem

## Related Issue
https://tyktech.atlassian.net/browse/TT-2687

## Motivation and Context
On remove certs, remove keys

## How This Has Been Tested
- Created 1 api with cert enabled
- Created via portal the key by pasting the cert
- Check that we have access
- Remove the cert via cert manager in dashboard
- Consume api... we should not have access
- Check in dashboard the key page, we should not see the key anymore
- Do the same but now in MDCB environment, once we remove the certificate then we should not have access to the api

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Review by matiasinsaurralde — APPROVED** [review]

**Comment by sredxny** [comment]
Added test coverage

**Review by tbuchaillot — APPROVED** [review]

File: gateway/api.go [hunk 1]
```diff
@@ -979,6 +980,10 @@ func keyHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 	case http.MethodDelete:
+		if certID != "" {
+			keyName =  generateToken(orgID, certID)
```
Comment (buger) [inline gateway/api.go:984]: I'm not quite sure why it requires a special case. Why we can't just standard "key" based on certificate fingerprint from the dashboard side?

Comment (sredxny) [reply gateway/api.go:984]: @buger makes sense. We do not longer need this PR. Made some changes in dashboard to take advantage of key deletion using only orgId+certId

**Comment by sredxny** [comment]
Closing this PR in favor to remove the keys using only the certID from the dashboard

**Pull request closed without merging**